templates/login.html

```diff
@@ -0,0 +1,134 @@
+           href='{{ authenticator_login_url | safe }}'>Sign in with {{ login_service }}</a>
+      </div>
+
+
+          <form action="{{ authenticator_login_url | safe }}"
+                method="post"
+                role="form">
+            <div class="auth-form-header">
+              <h1>Sign in</h1>
+            </div>
+            <div class='auth-form-body m-auto'>
+              <p id='insecure-login-warning' class='hidden alert alert-warning'>
+                Warning: JupyterHub seems to be served over an unsecured HTTP connection.
+                We strongly recommend enabling HTTPS for JupyterHub.
+              </p>
+
+              {% if login_error %}<p class="login_error">{{ login_error }}</p>{% endif %}
+              <input type="hidden" name="_xsrf" value="{{ xsrf }}" />
+              
+              <div class="feedback-container">
+                <input id="login_not_ok"
+                       type="button"
+                       class='btn btn-jupyter form-control hidden'
+                       value='Please log in to KBase to continue'
+                       tabindex="3" />
+                <input id="login_submit"
+                       type="submit"
+                       class='btn btn-jupyter form-control hidden'
+                       value='Log In with your KBase Account'
+                       tabindex="3" />
+                <div class="feedback-widget hidden">
+                  <i class="fa fa-spinner"></i>
+                </div>
+              </div>
+
+              {% block login_terms %}
+                {% if login_term_url %}
+                  <div id="login_terms" class="login_terms">
+                    <input type="checkbox"
+                           id="login_terms_checkbox"
+                           name="login_terms_checkbox"
+                           required />
+                    {% block login_terms_text %}
+                      {# allow overriding the text #}
+                      By logging into the platform you accept the <a href="{{ login_term_url }}">terms and conditions</a>.
+                    {% endblock login_terms_text %}
+                  </div>
+                {% endif %}
+              {% endblock login_terms %}
+            </div>
+          </form>
+      {% endblock login_container %}
+    </div>
+  {% endblock login %}
+{% endblock main %}
+{% block script %}
+  {{ super() }}
+  <script>
+    if (!window.isSecureContext) {
+      // unhide http warning
+      var warning = document.getElementById('insecure-login-warning');
+      warning.className = warning.className.replace(/\bhidden\b/, '');
+    }
+    // setup onSubmit feedback
+    $('form').submit((e) => {
+      var form = $(e.target);
+      form.find('.feedback-container>input').attr('disabled', true);
+      form.find('.feedback-container>*').toggleClass('hidden');
+      form.find('.feedback-widget>*').toggleClass('fa-pulse');
+    });
+
+    const enviromentOrigin = "https://ci.kbase.us"
```

Comment:
Can we make it dynamic? I can pass a env var something like `KB_ENV`, it will be either CI, APPDEV or PROD.

Reply:
Yup! This was just a placeholder

Reply:
Need to lookup how to get env variables into jinja and how to special case production, but I think it'll be straightforward